packages/project-access/src/file/file-access.ts

```diff
@@ -0,0 +1,52 @@
+import { promises as fs } from 'fs';
+import type { Editor } from 'mem-fs-editor';
+
+/**
+ * Read file asynchronously.
+ *
+ * @param path - path to file
+ * @param memFs - optional mem-fs-editor instance
+ * @returns - file content as string
+ */
+export async function readFile(path: string, memFs?: Editor): Promise<string> {
+    if (memFs) {
+        return memFs.read(path);
+    } else {
+        return fs.readFile(path, { encoding: 'utf8' });
+    }
+}
+
+/**
+ * Read JSON file asynchronously.
+ *
+ * @param path - path to JSON file
+ * @param memFs - optional mem-fs-editor instance
+ * @returns - file content as object of type T
+ */
+export async function readJSON<T>(path: string, memFs?: Editor): Promise<T> {
+    if (memFs) {
+        return memFs.readJSON(path) as unknown as T;
+    } else {
+        return JSON.parse(await readFile(path)) as T;
```

Comment:
Both these functions can throw errors if JSON parsing fails (mal-formed file, for example). Is it the responsibility of the caller to handle these? If so, can we add that to the JS Doc?

Reply:
This is a low level functionality and consumers need to handle errors. I've added this to JS Doc.